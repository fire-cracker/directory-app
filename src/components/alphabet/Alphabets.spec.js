import React from 'react';
import { shallow } from 'enzyme';

import Alphabet from './Alphabets';

describe('Tests for the Alphabet section', () => {
  const props = {
    activeContactEmail: 'Aogan.brown@example.com',
    activeAlphabet: 'A',
    contacts: [],
    onClickAlphabet: jest.fn()
  };
  let wrapper;

  beforeAll(() => {
    wrapper = shallow(<Alphabet {...props} />);
  });

  it('Should render the Alphabet section', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should click button', () => {
    wrapper
      .find('button')
      .at(0)
      .simulate('click');
    expect(props.onClickAlphabet).toHaveBeenCalled();
  });
});
