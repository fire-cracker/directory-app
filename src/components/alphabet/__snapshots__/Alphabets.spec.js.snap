// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for the Alphabet section Should render the Alphabet section 1`] = `
<Fragment>
  <button
    className="active"
    key="65"
    onClick={[MockFunction]}
    type="button"
    value="A"
  >
    A
    <sub>
      0
    </sub>
  </button>
  <button
    className={null}
    key="66"
    onClick={[MockFunction]}
    type="button"
    value="B"
  >
    B
    <sub>
      0
    </sub>
  </button>
  <button
    className={null}
    key="67"
    onClick={[MockFunction]}
    type="button"
    value="C"
  >
    C
    <sub>
      0
    </sub>
  </button>
  <button
    className={null}
    key="68"
    onClick={[MockFunction]}
    type="button"
    value="D"
  >
    D
    <sub>
      0
    </sub>
  </button>
  <button
    className={null}
    key="69"
    onClick={[MockFunction]}
    type="button"
    value="E"
  >
    E
    <sub>
      0
    </sub>
  </button>
  <button
    className={null}
    key="70"
    onClick={[MockFunction]}
    type="button"
    value="F"
  >
    F
    <sub>
      0
    </sub>
  </button>
  <button
    className={null}
    key="71"
    onClick={[MockFunction]}
    type="button"
    value="G"
  >
    G
    <sub>
      0
    </sub>
  </button>
  <button
    className={null}
    key="72"
    onClick={[MockFunction]}
    type="button"
    value="H"
  >
    H
    <sub>
      0
    </sub>
  </button>
  <button
    className={null}
    key="73"
    onClick={[MockFunction]}
    type="button"
    value="I"
  >
    I
    <sub>
      0
    </sub>
  </button>
  <button
    className={null}
    key="74"
    onClick={[MockFunction]}
    type="button"
    value="J"
  >
    J
    <sub>
      0
    </sub>
  </button>
  <button
    className={null}
    key="75"
    onClick={[MockFunction]}
    type="button"
    value="K"
  >
    K
    <sub>
      0
    </sub>
  </button>
  <button
    className={null}
    key="76"
    onClick={[MockFunction]}
    type="button"
    value="L"
  >
    L
    <sub>
      0
    </sub>
  </button>
  <button
    className={null}
    key="77"
    onClick={[MockFunction]}
    type="button"
    value="M"
  >
    M
    <sub>
      0
    </sub>
  </button>
  <button
    className={null}
    key="78"
    onClick={[MockFunction]}
    type="button"
    value="N"
  >
    N
    <sub>
      0
    </sub>
  </button>
  <button
    className={null}
    key="79"
    onClick={[MockFunction]}
    type="button"
    value="O"
  >
    O
    <sub>
      0
    </sub>
  </button>
  <button
    className={null}
    key="80"
    onClick={[MockFunction]}
    type="button"
    value="P"
  >
    P
    <sub>
      0
    </sub>
  </button>
  <button
    className={null}
    key="81"
    onClick={[MockFunction]}
    type="button"
    value="Q"
  >
    Q
    <sub>
      0
    </sub>
  </button>
  <button
    className={null}
    key="82"
    onClick={[MockFunction]}
    type="button"
    value="R"
  >
    R
    <sub>
      0
    </sub>
  </button>
  <button
    className={null}
    key="83"
    onClick={[MockFunction]}
    type="button"
    value="S"
  >
    S
    <sub>
      0
    </sub>
  </button>
  <button
    className={null}
    key="84"
    onClick={[MockFunction]}
    type="button"
    value="T"
  >
    T
    <sub>
      0
    </sub>
  </button>
  <button
    className={null}
    key="85"
    onClick={[MockFunction]}
    type="button"
    value="U"
  >
    U
    <sub>
      0
    </sub>
  </button>
  <button
    className={null}
    key="86"
    onClick={[MockFunction]}
    type="button"
    value="V"
  >
    V
    <sub>
      0
    </sub>
  </button>
  <button
    className={null}
    key="87"
    onClick={[MockFunction]}
    type="button"
    value="W"
  >
    W
    <sub>
      0
    </sub>
  </button>
  <button
    className={null}
    key="88"
    onClick={[MockFunction]}
    type="button"
    value="X"
  >
    X
    <sub>
      0
    </sub>
  </button>
  <button
    className={null}
    key="89"
    onClick={[MockFunction]}
    type="button"
    value="Y"
  >
    Y
    <sub>
      0
    </sub>
  </button>
  <button
    className={null}
    key="90"
    onClick={[MockFunction]}
    type="button"
    value="Z"
  >
    Z
    <sub>
      0
    </sub>
  </button>
</Fragment>
`;
